src/condor_startd.V6/boinc_mgr.cpp

```diff
@@ -196,12 +196,13 @@ BOINC_BackfillMgr::addSlot( BOINC_BackfillSlot* /*boinc_slot*/ )
 bool
 BOINC_BackfillMgr::rmSlot( int slot_id )
 {
-	if( ! m_slots[slot_id] ) {
+	size_t old_sz = m_slots.size();
+	auto slot_matches = [=](BackfillSlot* bs){bool match = slot_id == bs->getSlotID(); if (match) delete bs; return match;};
+	std::remove_if(begin(m_slots), end(m_slots), slot_matches);
+	if (m_slots.size() == old_sz) {
+		// No slot found/removed
```

Comment:
Note that `std::remove_if` doesn't actually remove, it just shuffle matches to the end of the sequence, and returns an iterator.  I think you want something like

```
auto it = std::remove_if(begin(m_slots), end(m_slots), slot_matches);
if (it == end(m_slots) {
    // no slots found
    return false;
} else {
   m.erase(it, end(it));
   ...
 ```

Reply:
Ah. remove_if() shuffles non-matches up, leaving holes at the end in an unspecified state. Looks like C++20 gives us erase_if(), which does what I was expecting. Your proposed change looks good.